### PR TITLE
Gitbook Docs

### PR DESCRIPTION
**Pull request by cullophid**
I have moved our docs to a gitbook

This makes them easier to read, and allows users to search for at function.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,10 @@
     "test": "babel-node node_modules/.bin/isparta cover _mocha -- --compilers js:babel-register 'src/**/_spec/*.js'",
     "posttest": "istanbul check-coverage --branches 100",
     "testBuild": "mocha buildTest.js",
-    "watch": "mocha --watch --compilers js:babel-register 'src/**/_spec/*.js'"
+    "watch": "mocha --watch --compilers js:babel-register 'src/**/_spec/*.js'",
+    "docs:prepare": "gitbook install",
```
Comment (cullophid) [inline package.json:17]: You can view the gitbook locally by running `npm run docs:watch`
The docs are hosted as a github page on the branch `gh-pages`

File: circle.yml [hunk 2]
```diff
@@ -15,3 +15,4 @@ deployment:
     commands:
       - echo -e "$NPM_USERNAME\n$NPM_PASSWORD\n$NPM_EMAIL" | npm login
       - npm publish
+      - ./bin/publish_docs
```
Comment (cullophid) [inline circle.yml:18]: Publishing docs is now part of the deploy step

**Comment by svozza** [comment]
Nice. I think it would look nicer if the heading names in the left-hand column were just the name of the function and the proper Haskell notation was the heading on the actual page with the info. Several of the type sigs are truncated as it is so it makes more sense to have them in the place with more screen real estate.

**Comment by cullophid** [comment]
I have added the changes from https://github.com/cullophid/date-fp/pull/24

File: docs/SUMMARY.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+  # Table of contents
+
+  * [Key concepts](docs/concepts.md)
+  * [Functions](docs/functions/index.md)
+    * [add :: String -> Number -> Date -> Date](docs/functions/add.md)
```
Comment (svozza) [inline docs/SUMMARY.md]: I think this should just be `[add]`, `[clone]` etc. with no type sig.

File: docs/functions/add.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+## add
```
Comment (svozza) [inline docs/functions/add.md]: If we moved the proper Haskell notation from Table of Contents to here, I think it would look better.

**Comment by cullophid** [comment]
@svozza I agree I have removed the type sig from the summary, and moved it to functions headers

**Comment by svozza** [comment]
Looks great and a definite improvement on my PR, we get nice links and informative type sigs with no losses.

**Comment by svozza** [comment]
Do you need to add gitbooks to the devDependencies?

**Comment by cullophid** [comment]
@svozza yes the gitbook get build as part of the deploy step

**Pull request merged**